File: JAVA/214. Shortest Palindrome.java
class Solution {
    public String shortestPalindrome(String s) {
        if(s == null || s.length() <= 1) {
            return s;
        }
        String rev_s = new StringBuilder(s).reverse().toString();
        int n = s.length();
        int i = 0;
        while(i < n) {
            if(s.substring(0, n - i).equals(rev_s.substring(i))) {
                break;
            }
            i++;
        }
        return rev_s.substring(0, i) + s;
    }
}

File: C++/214. Shortest Palindrome.cpp
class Solution {
public:
    string shortestPalindrome(string s) {
        int l = s.size();
        int i = 0;
        for(int j = l-1;j>=0;j--)
        {
            if(s[i]==s[j])
            {
                i++;
            }
        }
        if(i==l)
        {
            return s;
        }
        else
        {
            string temp = s.substr(i,l);
            reverse(temp.begin(),temp.end());
            return temp + shortestPalindrome(s.substr(0,i))+s.substr(i);
        }
    }
};
